### ISSUE TITLE
DWO does not seem to reconcile DevWorkspace after DevWorkspaceTemplate is changed

### ISSUE THREAD

**Issue by sleshchenko** [devfile/devworkspace-operator#432]
DWO does not seem to reconcile DevWorkspace after DevWorkspaceTemplate is changed.
P.S. DevWorkspaceTemplate seems to have the right ownerReference set.

Steps to reproduce:
1. Run DevWorkspace with Eclipse Che.
2. Update theia image in DevWorkspaceTemplate of Theia.
Expected: DevWorkspace deployment is updated with newer image.
Actual: DevWorkspace deployment uses the old configuration.

**Comment by sleshchenko** [comment]
Reproduced

**Comment by amisevsk** [comment]
> Expected: DevWorkspace deployment is updated with newer image.

I'd argue that this is a bad idea -- if there are a hundred DevWorkspaces, any update to the Theia DWT would result in >100 reconciles immediately being queued up, and potentially every DevWorkspace on the cluster restarting.

**Comment by sleshchenko** [comment]
Agreed that we don't won't to support it. We assign ownerRef for DWT being removed after DW is removed, but we don't want DWT updates - invoke reconcile loop
